src/Avalonia.Visuals/Rendering/SceneGraph/IVisualNode.cs

```diff
@@ -43,6 +43,11 @@ public interface IVisualNode : IDisposable
         /// Gets the layout bounds for the node in global coordinates.
         /// </summary>
         Rect LayoutBounds { get; }
+        
+        /// <summary>
+        /// Gets the original visual size before transformations
+        /// </summary>
+        Size VisualSize { get; }
```

Comment:
As far as I can tell, this is only needed by `IRenderTimeCriticalVisual`, is that right? Could this property be added to that interface? If it really needs to be added here it might be best to call it "PreTransformSize" or something?